lib/sproxyd.js

```diff
@@ -32,7 +34,19 @@ function _createRequest(req, log, callback) {
             return callback(error);
         }
         return callback(null, response);
-    }).on('error', callback);
+    }).on('error', err => {
+        // If callback was already called, log the error, but do not
+        // call the callback a second time.
+        if (callbackCalled) {
+            log.error('got socket error after response', {
```

Comment:
we will still get an error with this code:

make the request
get the response → set callbackCalled = true (line 23)
8 seconds later, timeout error → go in here and display an (unexpected) error
should the logic not be like this?

```
if (!callbackCalled) {
   callbackCalled = true;
   return callback(err);
}
if (err.code !== 'ERR_SOCKET_TIMEOUT') {
   log.error('got socket error after response', {err});
}
```

Reply:
I agree, and updated (+squashed) in https://github.com/scality/sproxydclient/pull/299/files#diff-ad65667505089c66c08316b902cbc633a991d4fc08e27c6e5995097678d0a262R38